app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/device/DeviceFragment.java

```diff
@@ -263,14 +243,9 @@ public boolean checkAppInstalled(String packageName) {
      */
     public void detectBackupEnabled() {
         totalTests++;
-        try {
-            PackageInfo packageInfo;
-            packageInfo = context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
-            if ((packageInfo.applicationInfo.flags & ApplicationInfo.FLAG_ALLOW_BACKUP) != 0) {
-                setDetected(allowBackup, R.string.allow_backup_detected_positive);
-            }
-        } catch (PackageManager.NameNotFoundException e) {
-            e.printStackTrace();
+        SecurityCheckResult result = securityService.check(SecurityCheckType.ALLOW_BACKUP_ENABLED);
+        if (result.passed()) {
+            setDetected(allowBackup, R.string.allow_backup_detected_positive);
```

Comment:
@TommyJ1994 Quick question. Is this the right way around (taken from the SDK example app). If backup is enabled that's a bad thing? I thought it'd be a good thing

Reply:
@aidenkeating  Yes, its a bad thing in the sense that you can retrieve application data without root access.

Reply:
@TommyJ1994 Nevermind, found information in https://github.com/feedhenry/mobile-security, very useful